src/main/java/org/ow2/proactive/addons/ldap_query/LDAPClient.java

```diff
@@ -78,40 +77,26 @@
 
     protected DirContext ldapConnection;
 
-    public LDAPClient(String ldapUrl, String ldapUsername, String ldapPassword, String ldapSearchBase,
-            String ldapSearchFilter, String ldapSelectedAttributes) {
-        allLDAPClientParameters.put(ARG_URL, ldapUrl);
-        allLDAPClientParameters.put(ARG_USERNAME, ldapUsername);
-        allLDAPClientParameters.put(ARG_PASSWORD, ldapPassword);
-        allLDAPClientParameters.put(ARG_SEARCH_BASE, ldapSearchBase);
-        allLDAPClientParameters.put(ARG_SEARCH_FILTER, ldapSearchFilter);
-        allLDAPClientParameters.put(ARG_SELECTED_ATTRIBUTES, ldapSelectedAttributes);
-    }
-
     public LDAPClient(Map<String, Serializable> actualTaskVariables, Map<String, Serializable> credentials) {
         ImmutableList<String> taskVariablesList = ImmutableList.of(ARG_URL,
+                                                                   ARG_DN_BASE,
                                                                    ARG_USERNAME,
                                                                    ARG_PASSWORD,
                                                                    ARG_SEARCH_BASE,
                                                                    ARG_SEARCH_FILTER,
                                                                    ARG_SELECTED_ATTRIBUTES);
 
         for (String variableName : taskVariablesList) {
-            Map<String, Serializable> mapWithVariables;
-            if (actualTaskVariables.containsKey(variableName)) {
-                mapWithVariables = actualTaskVariables;
-            } else {
-                mapWithVariables = credentials;
-            }
-            setLdapClientFields(mapWithVariables, variableName);
+            Serializable value = credentials.getOrDefault(variableName, actualTaskVariables.get(variableName));
+            setLdapClientFields(variableName, value);
         }
     }
 
-    private void setLdapClientFields(Map<String, Serializable> taskVariablesMap, String variableName) {
-        if (!taskVariablesMap.containsKey(variableName)) {
+    private void setLdapClientFields(String variableName, Serializable value) {
+        if (value == null) {
             throw new IllegalArgumentException("The missed argument for LDAPClient, variable name: " + variableName);
         }
-        allLDAPClientParameters.put(variableName, (String) taskVariablesMap.get(variableName));
+        allLDAPClientParameters.put(variableName, (String) value);
```

Comment:
This could be simplified by 
"allLDAPClientParameters.put(variableName, 
(String) Optional.ofNullable(value).orElseThrow(() -> new IllegalArgumentException("The missed argument for LDAPClient, variable name: " + variableName));"